Jellyfin.Server.Implementations/Item/ChapterRepository.cs

```diff
@@ -0,0 +1,123 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Jellyfin.Data.Entities;
+using MediaBrowser.Controller.Chapters;
+using MediaBrowser.Controller.Drawing;
+using MediaBrowser.Model.Dto;
+using MediaBrowser.Model.Entities;
+using Microsoft.EntityFrameworkCore;
+
+namespace Jellyfin.Server.Implementations.Item;
+
+/// <summary>
+/// The Chapter manager.
+/// </summary>
+public class ChapterRepository : IChapterRepository
+{
+    private readonly IDbContextFactory<JellyfinDbContext> _dbProvider;
+    private readonly IImageProcessor _imageProcessor;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="ChapterRepository"/> class.
+    /// </summary>
+    /// <param name="dbProvider">The EFCore provider.</param>
+    /// <param name="imageProcessor">The Image Processor.</param>
+    public ChapterRepository(IDbContextFactory<JellyfinDbContext> dbProvider, IImageProcessor imageProcessor)
+    {
+        _dbProvider = dbProvider;
+        _imageProcessor = imageProcessor;
+    }
+
+    /// <inheritdoc cref="IChapterRepository"/>
+    public ChapterInfo? GetChapter(BaseItemDto baseItem, int index)
+    {
+        return GetChapter(baseItem.Id, index);
+    }
+
+    /// <inheritdoc cref="IChapterRepository"/>
+    public IReadOnlyList<ChapterInfo> GetChapters(BaseItemDto baseItem)
+    {
+        return GetChapters(baseItem.Id);
+    }
+
+    /// <inheritdoc cref="IChapterRepository"/>
+    public ChapterInfo? GetChapter(Guid baseItemId, int index)
+    {
+        using var context = _dbProvider.CreateDbContext();
+        var chapter = context.Chapters.AsNoTracking()
+            .Select(e => new
+            {
+                chapter = e,
+                baseItemPath = e.Item.Path
+            })
+            .FirstOrDefault(e => e.chapter.ItemId.Equals(baseItemId) && e.chapter.ChapterIndex == index);
+        if (chapter is not null)
+        {
+            return Map(chapter.chapter, chapter.baseItemPath!);
+        }
+
+        return null;
+    }
+
+    /// <inheritdoc cref="IChapterRepository"/>
+    public IReadOnlyList<ChapterInfo> GetChapters(Guid baseItemId)
+    {
+        using var context = _dbProvider.CreateDbContext();
+        return context.Chapters.AsNoTracking().Where(e => e.ItemId.Equals(baseItemId))
+            .Select(e => new
+            {
+                chapter = e,
+                baseItemPath = e.Item.Path
+            })
+            .ToList()
```

Comment:
Is this even necessary? Can't EFCore map directly without ToList or AsEnumerable? If not, then shouldn't we use AsEnumerable?